### PR TITLE
[FRONTEND] Vendor page

### LINKED ISSUE #358: [FRONTEND] Vendor Page Design and Implementation
Vendor page design needs to be done immediately. 

Vendors will have a separate page to display their fundamental information (name, description, products, rating)
Vendor page should be editable by the owner vendor but no other people can edit it.
Vendor page displays the vendor's products in horizontal products component.
Vendor page header displays basic information.

I made initial Figma design like this:
image

### PR DESCRIPTION
**Pull request by abdullahyldz**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other

# Description

Vendor's profile page, which is accessible to both customers and vendors.
VendorHomePage is consisted of two components VendorSplash and TabsPane that consists UserReviews and VendorMainContent. UserReviews will show the reviews of the vendor that customers make after they make an order. VendorMainContent will display all the products of the vendor.
VendorSplash shows some information about vendor like 'title', 'description', 'profile picture' and 'rating' on marble background picture.

Page is designed to be able to edited by its owner. No other person can edit it. First check for owner is at splash, which shows Edit Mode switch to owner and Send Message button to others. 
Also, product cards get another props called 'EditMode' to become to its editable version. Edit Mode in the splash is needed to be on to make products editable, also product cards check for the current user to be the owner of the product.

# Screenshots

In case your pull request does not consist of visual changes, you can delete this section. 
In customer and other vendor's eyes:
image
In vendor owner's eyes
image
Some background, border, margin changes:
image

# Test results

Please run all the tests and dump the result/summary here in a code block.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Potential problems/Important remarks

In the second image, you can see some products are not in edit mode because they are other vendor's products. Normally, they shouldn't be in this page. 
Also, I am signing in as Can Batuk which has id like 55 or something but the products who has vendor Can Batuk has id 3. So, I played in /init method to access edit mode. We should learn the credentials from backend.

/closes #358

### CODE REVIEW

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: Since you use `onChange` shouldnt you also assign `value={editable}`?

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: Not sure if Button has href but in any case a regular html link shouldn't be used. You need to use either Link from react-router or have a onClick which uses `history.push('/profile/messages/`). 

Also messages page does not allow creating a new conversations. For this you will need to use a  "new message modal". We are working on it with @ozdenizdolu in F-prepare-vendor-ui branch.

File: ecommerce/frontend/src/components/layout/Content.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/layout/Content.js]: the reason we have the `location.pathname` is because when you are in the profile age the notification will appear twice, both in Content and in Profile page. maybe you can add those `isUser...` conditions?

File: ecommerce/frontend/src/components/layout/Header.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/layout/Header.js:88]: for that previous button i mentioned (the one with href), maybe you can use the same code here?

File: ecommerce/frontend/src/components/pages/VendorHomepage.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/pages/VendorHomepage.js]: Please implement this (at least using mock) before merging the pr

File: ecommerce/frontend/src/components/pages/VendorHomepage.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/pages/VendorHomepage.js]: shouldn't default value of editMode be the value of `isVendorAndOwner`?

File: ecommerce/frontend/src/components/pages/VendorHomepage.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/pages/VendorHomepage.js]: what is this used for?

File: ecommerce/frontend/src/components/product_card/ProductCard.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/product_card/ProductCard.js]: Please implement these in this PR

File: ecommerce/frontend/src/components/product_card/ProductCard.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/product_card/ProductCard.js]: should be `const`

File: ecommerce/frontend/src/components/product_card/ProductCard.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/product_card/ProductCard.js]: Can you make currency TL instead just like we used everywhere?

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: don't forget to remove this once done

**Review by MehdiSaffar — CHANGES_REQUESTED** [review]
I like the design change and I overall think the code is solid.

Please add some mock handlers in mock/handlers.js so that this can be tested with backend.

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: Switch component handles its own state.

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: Ok, then I will use onClick function, which first opens a modal for 'Sending a Message', after sending a message function uses 
`history.push('/profile/messages/').` and finishes.

File: ecommerce/frontend/src/components/product_card/ProductCard.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/product_card/ProductCard.js]: sure

File: ecommerce/frontend/src/components/pages/VendorHomepage.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/pages/VendorHomepage.js]: deleted

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: I added mock data. These are gone, now.

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (furkan-varol) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: Unnecessary console.log

**Review by furkan-varol — COMMENTED** [review]
I agree with most of the comments made by Mehdi. Solid work overall, thanks for your effort. After the missing functionalities are implemented, I will fully test and approve.

File: ecommerce/frontend/src/components/layout/Content.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/layout/Content.js]: I hadn't seen this component.
image

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (ozdenizdolu) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: I might be missing something but shouldn't this also make a message modal visible? I couldn't see a message modal anywhere in the code.

File: ecommerce/frontend/src/components/pages/VendorHomepage.js
Comment (ozdenizdolu) [inline ecommerce/frontend/src/components/pages/VendorHomepage.js]: should this be commented?

**Review by ozdenizdolu — COMMENTED** [review]
Nicely written. Good job. I will approve after we resolve my comments.

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: Real endpoint return the data directly in response, not within data field

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: I agree with @ozdenizdolu, please check how it is done in Order component

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: No need to create a function, you can pass `onClick={onSendMessage}` directly

File: ecommerce/frontend/src/components/product_card/ProductCard.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/product_card/ProductCard.js]: Not sure what this is necessary for? I never had to use it personally.

File: ecommerce/frontend/src/mocks/handlers.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/mocks/handlers.js]: Return vendorDetails[minId] directly in the response like the backend doe

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: In the updated master branch I exported this function from utils. Please use it from there

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (MehdiSaffar) [inline ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: make sure vendorId is passed as an integer from parent component instead (to avoid subtle bugs where you forget it is a string)

**Review by MehdiSaffar — CHANGES_REQUESTED** [review]
Good work, needs to be perfected.

File: ecommerce/frontend/src/components/pages/VendorHomepage.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/pages/VendorHomepage.js]: Commented out, just because I don't have a legit Vendor account with its products. It gets 500 error, I think because of invalid token usage

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: I fixed it, I will also test it after Backend finalize VendorDetail endpoint.

File: ecommerce/frontend/src/components/product_card/ProductCard.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/product_card/ProductCard.js]: It is left from the time you first mentioned about the Date format that Backend sends. I had imported it then, but it is unnecessary now.

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: I see MessageModal in Purchase.js  
In that file, there is no history.push  like operations. So, I have removed the history.push for now. I didn't get the flow of messaging, should we direct customer to profile/messages after they successfully send a message?

File: ecommerce/frontend/src/mocks/handlers.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/mocks/handlers.js]: done

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: done

File: ecommerce/frontend/src/components/VendorContent/VendorSplash.js
Comment (abdullahyldz) [reply ecommerce/frontend/src/components/VendorContent/VendorSplash.js]: done

**Comment by abdullahyldz** [comment]
Added popconfirm to delete product.
Message modal opens and closes. Modal shows the vendor title as receiver.
VendorId is parsed to integer.
Removed pagination tools from Vendor reviews.
Vendor details gets value from Backend. However, Backend is not complete.

**Comment by MehdiSaffar** [comment]
@ozdenizdolu and I have the following commments:

- Add a vendor page shortcut (for example in header bar and/or in profile menu)
- It would be better to show edit/delete buttons like in Card or Address
- Either need to implement pagination or increase pageSize to a big number (like 1000), some products are not shown
- Show a spinner in the delete button of a product + refresh the product list so that the product is no longer seen
- Refresh the product list when a product is edited to show changes
- Show notifications for edit/delete successful?
- Allow editing title description profile picture
- Need to divide total_rating / rating_count
- Search products shows product from various vendors, need to add {vendor: vendorId} filter in /search/products
- Products should be sorted by creation/modification date.

**Comment by MehdiSaffar** [comment]
We did a long collaborative session over Google meets and vscode Live share and we made all the requested changes as well as polished the vendor home page. We can merge.

**Review by MehdiSaffar — APPROVED** [review]

**Pull request merged**
